### ISSUE TITLE
Add a warning when code tries to access res://

### ISSUE THREAD

**Issue by skyace65** [godotengine/godot-proposals#3755]
### Describe the project you are working on

Godot Documentation.

### Describe the problem or limitation you are having in your project

Adding a FAQ to the exporting docs was recently proposed in [this issue](https://github.com/godotengine/godot-docs/issues/5272). One of two points for the FAQ was that projects should not access `res://` as it breaks on export. People are already told not to do this in the step by step section, file and dictionary class reference pages. But apparently this still comes up frequently in the discord help channel.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

 Instead of adding another message about it in the docs which could easily be ignored as well I suggest adding an engine warning when a script tries to access `res://`, an exception would need to be made for tools scrips since it's okay for plugins to do this. This would be significantly harder for people to miss, means we don't need to add more to the docs, and could cut back on the amount of people asking for help.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Calinou said in the doc issue that this code could be used to check if a tool script is accessing `res://` or not before printing the warning, `Engine::get_singleton()->is_editor_hint()`.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

It's a warning

**Comment by zinnschlag** [comment]
To clarify: With access you mean write-access, right?

**Comment by Calinou** [comment]
> To clarify: With access you mean write-access, right?

`res://` read-only can also break in an exported project, but only in some cases. It's harder to detect automatically.

**Comment by zinnschlag** [comment]
Right. With write-access generally being wrong in exported projects, a warning makes sense. But with read-access only being sometimes wrong a warning would produce a lot of false positives. Not ideal.

**Comment by Xrayez** [comment]
Unless there's a way to disable the warning, I'm against this. There will be a lot of false positives and will worsen Godot's image as a professional tool because of this. See for instance godotengine/godot#24222.

**Comment by Calinou** [comment]
> Unless there's a way to disable the warning, I'm against this. There will be a lot of false positives and will worsen Godot's image as a professional tool because of this. See for instance [godotengine/godot#24222](https://github.com/godotengine/godot/issues/24222).

There will be a project setting to disable this warning.
